sound/soc/sof/Kconfig

```diff
@@ -50,6 +50,25 @@ config SND_SOC_SOF_DEBUG_PROBES
 	  Say Y if you want to enable probes.
 	  If unsure, select "N".
 
+config SND_SOC_SOF_CLIENT
+	tristate
+	select VIRTUAL_BUS
+	help
+	  This option is not user-selectable but automagically handled by
+	  'select' statements at a higher level
+
+config SND_SOC_SOF_CLIENT_SUPPORT
+	bool "SOF enable clients"
+	depends on SND_SOC_SOF
+	help
+	  This adds support for client support with Sound Open Firmware.
+	  The SOF driver adds the capability to separate out the debug
+	  functionality for IPC tests, probes etc. into separate client
+	  devices. This option would also allow adding client devices
+	  based on DSP FW capabilities and ACPI/OF device information.
+	  Say Y if you want to enable clients with SOF.
+	  If unsure select "N".
```

Comment:
Sorry, should have asked earlier, but I'm still not quite convinced why we even need this option exposed. We will have several functions like probes, IPC flood tests etc. that all depend on this. But each of those options has its own Kconfig entry, right? Wouldn't it be better to just have those entries and let them select CLIENT? You also said that typically distros won't enable this, which I still don't quite understand why.

Reply:
No, we want to avoid blanket selections and make sure there is the ability to cherry-pick which options are selected. It's the job of distributions or users to make the choices, not ours.

Reply:
@plbossart I understand what you mean and in general I'm certainly in favour of flexibility and user choice, but this option just seems to "internal" to me. Like you don't ask the user whether they want to support topology parsing. Tha analogy isn't very good, but what I mean is - the used wants to decide whether they need, say, IPC flooding. And they don't care if it depends on client support or some other "implementation detail." But maybe my vision of the client support role isn't right.